### PR TITLE
Updates Valhalla, adds missing items/features

### PR DESCRIPTION
**Pull request by RipGrayson**
## About The Pull Request

Updates Valhalla so that it has;
- The ability to spawn xenos of any maturity.
- The ability to spawn kings, queens and gorgers.
- A vendor with all seasonal guns in it, regardless of season.
- Xeno HUDs in firing range
- Various missing weapons, including rocket sledge, harvester spear etc.
- Access to Leicester ammo and ammo for disposable rocket launchers.

Addresses https://github.com/tgstation/TerraGov-Marine-Corps/issues/9653 and https://github.com/tgstation/TerraGov-Marine-Corps/issues/9783
## Why It's Good For The Game
Valhalla just needs a little bit of updating to fix some missing items that either weren't present or were overlooked during the initial PR.
Also fixes for issues good.

## Changelog
:cl:
qol: Made it so marines can now spawn xenos of any maturity in Valhalla
qol: Valhalla now has xeno HUDs scattered around the firing ranges.
qol: Valhalla now has a special vendor for testing seasonal weapons.
qol: Updated Valhalla vendors to have Harvester spear and rocket sledge.
fix: Fixed marines being unable to spawn queens, kings, and gorgers in Valhalla.
fix: Fixed ammo for some weapon types not being in Valhalla vendors
/:cl:

### CODE REVIEW

**Comment by private-tristan** [comment]
can you add combat pack to the vendor? doesn't seem to be an option there despite being a req item

**Comment by RipGrayson** [comment]
> can you add combat pack to the vendor? doesn't seem to be an option there despite being a req item

Sure, just added it to Valhalla req vendor.

**Comment by Araulius** [comment]
thanks grayson. this makes me happy.

**Comment by Technical-l** [comment]
hey can you add ammo to the disposable rocket launchers in valhalla? when you get them from the req vendor, they spawn with no ammo

File: code/_globalvars/lists/mobs.dm [hunk 1]
```diff
@@ -48,24 +48,103 @@ GLOBAL_LIST_EMPTY(roundstart_species)
 GLOBAL_LIST_EMPTY_TYPED(xeno_caste_datums, /list/datum/xeno_caste)
 GLOBAL_LIST_INIT(all_xeno_types, list(
 	/mob/living/carbon/xenomorph/runner,
+	/mob/living/carbon/xenomorph/runner/mature,
+	/mob/living/carbon/xenomorph/runner/elder,
```
Comment (TiviPlus) [inline code/_globalvars/lists/mobs.dm:52]: why doesnt this just use a typecache

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/_globalvars/lists/mobs.dm (hunk 1 above)
Comment (RipGrayson) [reply code/_globalvars/lists/mobs.dm:52]: The biggest problem is that you can spawn non-admeme xenos in Valhalla if you did a strict typecache, as well as xenos that could be used to grief/lag the server like xenos with functional AI.

**Comment by RipGrayson** [comment]
If a typecache is still required I might need to think about the best to do it. Local testing revealed a bunch of ways to metagame or make valhalla unusable with typecache enabled.

**Comment by OpsCoretango** [comment]
add please shoulder attachment

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**